lib/classes/Service.js

```diff
@@ -113,7 +113,12 @@ class Service {
         that.custom = serverlessFile.custom;
         that.plugins = serverlessFile.plugins;
         that.resources = serverlessFile.resources;
-        that.functions = serverlessFile.functions || {};
+        that.functions = {};
+
+        // normalize functions with `.` in name to `-dot-`
+        _.forIn(serverlessFile.functions, (value, key) => {
+          that.functions[key.replace(/[.]/g, '-dot-')] = value;
```

Comment:
I feel this it's not elegant, any ideas where I should move that? Is it even right direction?